extensions/extensions-configuration-store-vault/src/main/java/com/boozallen/aissemble/configuration/dao/VaultPropertyDao.java

```diff
@@ -50,14 +50,18 @@ public boolean checkEmpty() {
      * @param groupName to read from the vault
      * @param propertyName to read from the vault
      */
-    @Override
+    @Override 
     public Property read(String groupName, String propertyName) {
         try {
             final Vault vault = getVault();
             // read from Vault
             final LogicalResponse readResponse = vault.logical()
                     .read(String.format("aissemble-properties/%s/%s", groupName, propertyName));
-
+            // Check for null responses
+            if (readResponse == null || readResponse.getData() == null || readResponse.getData().get(propertyName) == null) {
+                logger.warn("Property not found: {}/{}", groupName, propertyName);
+                return null;
+            }
```

Comment:
Q: I thought exception will be thrown if server can't find the property?

Reply:
Yep. However, at the config init process, if ConfigStoreInit can't find the property when reading "fully-loaded" status(since it wasn't loaded previously), we don't want it to directly fail, but rather execute write.

Reply:
Could we retain the original logic of throwing an exception when a property is not found, and instead just utilize a try/catch block in the `isFullyLoaded()` function to handle that exception?

Reply:
yep! I've reverted it back and use a try/catch in `isFullyLoaded`